### PR TITLE
Account management

### PR DESCRIPTION
**Pull request by yramanchuk**
Implementation of Account Management.
UI mocks can be provided by request.

Pull request contains:
- new UI flow for account management;
- refactoring of i18n;

Thing to be done:
- correct some of the labels to fit mocks;
- implement swift sample;
- add UI and unit tests;
- implement Account management callbacks;

### CODE REVIEW

File: FirebaseAuthUI/FUIAuth.h [hunk 1]
```diff
@@ -53,6 +55,10 @@ typedef void (^FUIAuthResultCallback)(FIRUser *_Nullable user, NSError *_Nullabl
 
 @optional
 
+// TODO: add docs
```
Comment (morganchen12) [inline FirebaseAuthUI/FUIAuth.h]: Please address this before merging

File: FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperationForgotPassword.m [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  if (![[FUIAuthBaseViewController class] isValidEmail:email]) {
+    [self showAlertWithMessage:FUILocalizedString(kStr_InvalidEmailError)];
+    return;
+  }
+
+  [_delegate incrementActivity];
+
+  [_delegate.auth sendPasswordResetWithEmail:email
+                             completion:^(NSError *_Nullable error) {
+     // The dispatch is a workaround for a bug in FirebaseAuth 3.0.2, which doesn't call the
```
Comment (XiangtianDai) [inline FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperationForgotPassword.m]: This should be fixed by now. Is there any chance we can make FirebaseUI depend on FirebaseAuth ~> 3.1.0 ?

File: FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  return [classMap[NSStringFromClass([self class])] integerValue];
+}
+
+#pragma mark - protected methods
+
+- (void)finishOperationWithError:(nullable NSError *)error {
+  if (error) {
+    switch (error.code) {
+      case FIRAuthErrorCodeEmailAlreadyInUse:
+        [self showAlertWithMessage:FUILocalizedString(kStr_EmailAlreadyInUseError)];
```
Comment (XiangtianDai) [inline FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m]: So the `authUI:didFinishOperation:error:` delegate method is not called in error case?

File: FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperationForgotPassword.m [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+       [_delegate decrementActivity];
+
+       if (error) {
+         [self finishOperationWithError:error];
+         return;
+       }
+
+       NSString *message = [NSString stringWithFormat:
+           FUILocalizedString(kStr_PasswordRecoveryEmailSentMessage), email];
+       [self showAlertWithMessage:message];
```
Comment (XiangtianDai) [inline FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperationForgotPassword.m]: `finishOperationWithError:`?

If so, the alert won't be dismissed when the delegate receives the `authUI:didFinishOperation:error:` call, is that okay?

File: FirebaseAuthUI/FUIStaticContentTableViewManager.m [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+  _contents = contents;
+  [self.tableView reloadData];
+}
+
+- (void)setTableView:(nullable UITableView *)tableView {
+  _tableView = tableView;
+  [tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:kCellReuseIdentitfier];
+
+  UINib *passwordCellNib = [UINib nibWithNibName:NSStringFromClass([FUIPasswordTableViewCell class])
+                                  bundle:[FUIAuthUtils frameworkBundle]];
```
Comment (XiangtianDai) [inline FirebaseAuthUI/FUIStaticContentTableViewManager.m]: Please align colons.

File: FirebaseAuthUI/FUIStaticContentTableViewManager.m [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+  [tableView registerNib:passwordCellNib forCellReuseIdentifier:kPasswordCellReuseIdentitfier];
+
+  UINib *inputCellNib = [UINib nibWithNibName:NSStringFromClass([FUIInputTableViewCell class])
+                                  bundle:[FUIAuthUtils frameworkBundle]];
```
Comment (XiangtianDai) [inline FirebaseAuthUI/FUIStaticContentTableViewManager.m]: Please align colons.

File: FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+  }
+
+  [[FUIAuth defaultAuthUI] invokeOperationCallback:[self operationType] error:error];
+}
+
+- (void)showSelectProviderDialog:(nullable FUIAccountSettingsChooseProviderHandler)handler
+                      alertTitle:(nullable NSString *)title
+                    alertMessage:(nullable NSString *)message
+                alertCloseButton:(nullable NSString *)closeActionTitle {
+  UIAlertController *alert =
```
Comment (morganchen12) [inline FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m]: FirebaseUI only supports 8+, so this is not an issue. We dropped iOS 7 when integrating Twitter auth.

File: FirebaseAuthUI/AccountManagement/FUIAccountSettingsViewController.m [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+/** @var kUserAccountImage
+    @brief Name of icon to show default user account.
+ */
+static NSString *const kUserAccountImage = @"ic_account_circle.png";
+
+@interface FUIAccountSettingsViewController () <FUIAccountSettingsOperationUIDelegate>
+@end
+
+@implementation FUIAccountSettingsViewController {
+  __unsafe_unretained IBOutlet UITableView *_tableView;
```
Comment (morganchen12) [inline FirebaseAuthUI/AccountManagement/FUIAccountSettingsViewController.m]: Adding to this, if the IBOutlet is owned entirely by the view controller (which it should be), you can go ahead and make this a strong reference.

**Comment by cerupcat** [comment]
Is account management available? I see it checked in, but it's commented out of the example project and not included in cocoapods checkout. @morganchen12 @yramanchuk

**Pull request merged**